ddtrace/tracer/context.go

```diff
@@ -20,7 +20,9 @@ func SpanFromContext(ctx context.Context) (Span, bool) {
 	}
 	s := opentracing.SpanFromContext(ctx)
 	if s != nil {
-		return s.(ddtrace.Span), true
+		if span, ok := s.(ddtrace.Span); ok {
+			return span, true
+		}
 	}
 	return &ddtrace.NoopSpan{}, false
```

Comment:
this would be the more "idiomatic" way of doing it

```suggestion
	span, ok := s.(ddtrace.Span)
	if !ok {
		return &ddtrace.NoopSpan{}, false
	}
	return span, true
```

1. https://github.com/golang/go/wiki/CodeReviewComments#indent-error-flow
2. `nil` check is not needed if you do the "comma, ok" type assertion

Reply:
Oh, absolutely, thanks for the suggestion!